src/main/java/org/synyx/urlaubsverwaltung/statistics/vacationoverview/web/ApplicationForLeaveVacationOverviewViewController.java

```diff
@@ -47,25 +79,217 @@ public String applicationForLeaveVacationOverview() {
 
     @PreAuthorize(IS_PRIVILEGED_USER)
     @GetMapping("/vacationoverview")
-    public String applicationForLeaveVacationOverview(Model model) {
-        Person signedInUser = personService.getSignedInUser();
+    public String applicationForLeaveVacationOverview(
+        @RequestParam(required = false) Integer year,
+        @RequestParam(required = false) String month,
+        @RequestParam(required = false) String department,
+        Model model, Locale locale)
+    {
+        final var signedInUser = personService.getSignedInUser();
+        final var departments = departmentService.getAllowedDepartmentsOfPerson(signedInUser);
+        model.addAttribute("departments", departments);
+
+        final var fallbackDepartment = departments.isEmpty() ? "" : departments.get(0).getName();
+        final var selectedDepartmentName = hasText(department) ? department : fallbackDepartment;
+        model.addAttribute("selectedDepartment", selectedDepartmentName);
 
-        prepareDepartments(signedInUser, model);
+        final var startDate = getStartDate(year, month);
+        final var endDate = getEndDate(year, month);
 
         model.addAttribute("currentYear", ZonedDateTime.now(UTC).getYear());
-        model.addAttribute("currentMonth", ZonedDateTime.now(UTC).getMonthValue());
+        model.addAttribute("selectedYear", startDate.getYear());
+
+        String selectedMonth;
+        if (month == null) {
+            selectedMonth = String.valueOf(startDate.getMonthValue());
+        } else if (hasText(month)) {
+            selectedMonth = month;
+        } else {
+            selectedMonth = "";
+        }
+        model.addAttribute("selectedMonth", selectedMonth);
+
+        final var overviewPersons = getOverviewPersonsForUser(signedInUser, departments, selectedDepartmentName);
+        final var sickNotes = sickNoteService.getAllActiveByYear(year == null ? LocalDate.now(UTC).getYear() : year);
+
+        final var vacationsByEmail = new HashMap<String, List<Application>>();
+        for (Person p : overviewPersons) {
+            var apps = applicationService.getApplicationsForACertainPeriodAndPerson(startDate, endDate, p);
+            vacationsByEmail.put(p.getEmail(), apps);
+        }
+
+        var monthsByNr = new HashMap<Integer, AbsenceOverviewMonthDto>();
+
+        // `date` is increased by one month at the end of this outer loop
+        // the outer loop builds the absence overview view model for every user desired month (specific one or all 12 🐵)
+        var date = startDate;
```

Comment:
Könnte man ab hier ggfs gut auslagern damit es leserlicher wird und fachlich beschreiben?

Reply:
sehe ich im "großen refactoring" mit dabei.

dass die kalender datenstruktur in einem service erstellt wird. und der viewController nur noch eine datenstruktur zu dtos mapped. so irgendwie.